File: Silver III/15656/yeolyi.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().split('\n')

const [N, M] = input[0].split(' ').map(Number);
const arr = input[1].split(' ').map(Number).sort((a, b) => a-b);

let ret = ''

function solve(count, temp) {
    if (count === 0) { 
        ret += temp.join(' ');
        ret += '\n';
        return;
     }
     for (let i=0; i<N; i++) {
        solve(count-1, [...temp, arr[i]]);
     }
}

solve(M, []);
console.log(ret);

File: Silver II/1138/yeolyi.py
N = int(input())
arr = [int(x) for x in input().split()]
ans = [None for _ in range(N)]


def solve(i, idx, taller):
    if taller == 0 and ans[idx] is None:
        ans[idx] = i + 1
    else:
        if ans[idx] is None:
            solve(i, idx + 1, taller - 1)
        else:
            solve(i, idx + 1, taller)


for i in range(N):
    solve(i, 0, arr[i])

print(" ".join([str(x) for x in ans]))
